NuRadioMC/SignalProp/analyticraytracing.py

```diff
@@ -1584,23 +1627,37 @@ class describing the index-of-refraction profile
         self.__logger.debug("X2 - X1 = {}, X1r = {}, X2r = {}".format(self.__X2 - self.__X1, X1r, X2r))
         self.__x1 = np.array([X1r[0], X1r[2]])
         self.__x2 = np.array([X2r[0], X2r[2]])
-
         self.__logger.debug("2D points {} {}".format(self.__x1, self.__x2))
-        self.__r2d = ray_tracing_2D(self.__medium, self.__attenuation_model, log_level=log_level,
-                                    n_frequencies_integration=self.__n_frequencies_integration)
 
-    def set_solution(self, C0s, C1s, solution_types, reflection=None, reflection_case=None):
+    def set_solution(self, raytracing_results, i_shower, channel_id):
```

Comment:
is there a way to make the interface better? The `i_shower` and `channel_id` paramters are just used to index the `ray_tracing_results` dict. The ray tracer can also be used independent of a MC simulation. Can't you just pass a dict with the properties of this specific shower/channel combination?

Reply:
I don't think there is an easy way to do this. raytracing_results is a dictionary and simulation.py does not know which properties the raytracer needs. The only solution I can think of is to loop through all keys and copy the values into a new dictionary, but that would probably be slow.

Reply:
but you know which properties the ray tracer needs by calling the corresponding function of the ray tracer `get_output_parameters`.

Reply:
So we check if all the parameters from get_output_parameters are set? I'm okay with that, but it could be a problem if we have some parameter that is not set for every solution.

Reply:
we could also just check the first parameter to avoid this problem. The first parameter must then be the parameter that is always defined.

Reply:
Done